project/src/components/offer-details/offer-details.tsx

```diff
@@ -1,70 +1,126 @@
 import {Offer} from '../../types/offer';
-import {UserReview } from '../review/review';
-import {ReviewForm} from '../review-form/review-form';
+import ReviewForm from '../review-form/review-form';
 import {Map} from '../map/map';
 import {OffersNearList} from '../offers-near-list/offers-near-list';
+import {useEffect} from 'react';
+import {Loader} from '../loader/loader';
+import {ReviewList} from '../review-list/review-list';
+import {State} from '../../types/state';
+import {connect, ConnectedProps} from 'react-redux';
+import {chooseActiveOffer} from '../../store/actions';
+import {ThunkAppDispatch} from '../../types/actions';
 
 import {
   AppRoute,
-  CardClassType
+  AuthorizationStatus,
+  CardClassType,
+  DataStatus
 } from '../../common/const';
 import {
-  getCityOffers,
   getOfferCapacity,
   percentageRating
 } from '../../utils/utils';
 import {
+  Link,
   Redirect,
   useParams
 } from 'react-router-dom';
-import {State} from '../../types/state';
-import {connect, ConnectedProps} from 'react-redux';
-import {chooseActiveOffer} from '../../store/actions';
-import {Dispatch} from 'react';
-import {Actions} from '../../types/actions';
-
-const OFFERS_LENGTH_CONSTRAINT = 3;
+import {
+  fetchNearbyOffersAction,
+  fetchOfferCommentsAction,
+  fetchOfferDetailsAction
+} from '../../store/api-actions';
 
-const mapStateToProps = ({activeCity, offers, reviews, activeOffer}: State) => ({
-  activeCity,
-  offers,
-  reviews,
-  activeOffer,
-});
+const mapStateToProps = (
+  {
+    activeCity,
+    reviews,
+    activeOffer,
+    currentOffer,
+    nearbyOffers,
+    isOfferDetailsLoaded,
+    isNearbyOffersLoaded,
+    isOfferCommnetsLoaded,
+    authorizationStatus,
+  }: State) => (
+  {
+    activeCity,
+    reviews,
+    activeOffer,
+    currentOffer,
+    nearbyOffers,
+    isOfferDetailsLoaded,
+    isNearbyOffersLoaded,
+    isOfferCommnetsLoaded,
+    authorizationStatus,
+  }
+);
 
-const mapDispatchToProps = (dispatch: Dispatch<Actions>) => (
+const mapDispatchToProps = (dispatch: ThunkAppDispatch) => (
   {
     onOfferChoose(offer: Offer | null) {
       dispatch(chooseActiveOffer(offer));
     },
+    fetchOfferDetails(offerId: string) {
```

Comment:
```suggestion
    onFetchOfferDetails(offerId: string) {
```
коллбеки, что приходят в пропсы, должны начинаться с `on`